File: baekjoon/src/greedy/B_2212.java
package greedy;

import java.util.*;
import java.io.*;

public class B_2212 {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		int K = Integer.parseInt(br.readLine());
		
		int[] sensor = new int[N];
		StringTokenizer st = new StringTokenizer(br.readLine());
		for(int i=0;i<N;i++) 
			sensor[i] = Integer.parseInt(st.nextToken());
		
		Arrays.sort(sensor);
		
		int[] diff = new int[N - 1];
		for(int i=0;i<N-1;i++)
			diff[i] = sensor[i + 1] - sensor[i];
		
		Arrays.sort(diff);
		int answer = 0;
		
		for(int i=0;i<N-K;i++)
			answer += diff[i];
		
		System.out.println(answer);
	}

}

File: src/kotlin/greedy/B_2212.kt
fun main() = with(System.`in`.bufferedReader()) {
    val n = readLine().toInt()
    val k = readLine().toInt()
    val arr = readLine().split(" ").map { it.toInt() }.toIntArray()
    val diff = IntArray(n - 1)
    arr.sort()

    for (i in 1..<n) {
        diff[i - 1] = arr[i] - arr[i - 1]
    }
    diff.sort()
    var answer = 0
    for (i in 0..<n - k) {
        answer += diff[i]
    }

    print(answer)
}
